opm/parser/eclipse/EclipseState/IOConfig/IOConfig.hpp

```diff
@@ -158,6 +158,15 @@ namespace Opm {
 
         std::string getRestartFileName(const std::string& restart_base, int report_step, bool output) const;
 
+        std::string getDeckFileName();
```

Comment:
Well there is a `getBaseName()`method as well - in my opinion that should be the only required method.